roxie/roxiemem/roxiemem.cpp

```diff
@@ -748,10 +748,8 @@ static void *suballoc_aligned(size32_t pages, bool returnNullWhenExhausted)
                             //Check if all memory above this allocation is allocated - if so extend the HWM
                             if ((hbi == 0) && (i+1 == heapHWM))
                             {
-                                if (startHbi != 0)
-                                    heapHWM = start+1;
-                                else
-                                    heapHWM = start;
+                                //Could be some space in the first chunk - will be checked next time allocate is called.
```

Comment:
This one looks like a more significant change than the rest. What was coverity's complaint?

Reply:
The complain was that startHBi must be non-null (since hbi is non null).  I think the intent should be to check the updated version of the mask, but it is easier to check that next time.